skops/io/tests/test_persist.py

```diff
@@ -804,3 +804,30 @@ def test_loads_from_str():
     msg = "Can't load skops format from string, pass bytes"
     with pytest.raises(TypeError, match=msg):
         loads("this is a string")
+
+
+class _BoundMethodHolder:
+    """Used to test the ability to serialise and deserialize bound methods"""
+
+    def __init__(self, variant: str):
```

Comment:
I appreciate that you want to provide a somewhat realistic example here. But I think overall it detracts from what the test is actually testing, since a lot of the code is not actually used. This makes it harder for future readers of the code to understand what happens. Thus I would prefer to reduce the example to the minimum necessary. Maybe a comment or two in the test could also help.

Reply:
Good point, I can make this change and add some comments. 

I know most of the tests in this repo are just at the base level of the `test_...` file, would it be worth adding in a `TestBoundMethodSerialization` to go over edge cases, like the infinite loop caused by the original issue?

Reply:
Personally I don't mind a class containing multiple related tests, I've added a few of those myself. Since they were accepted, I assume the other maintainers don't mind either ;)